tests/integration/connection_manager_translate_test.py

```diff
@@ -0,0 +1,137 @@
+import uuid
+
+from mock import patch
+
+from hazelcast import HazelcastClient
+from hazelcast.core import Address, MemberInfo, MemberVersion, EndpointQualifier, ProtocolType
+from hazelcast.errors import IllegalStateError, TargetDisconnectedError
+from tests.base import HazelcastTestCase
+
+_UNREACHABLE_ADDRESS = Address("192.168.0.1", 5701)
+_MEMBER_VERSION = MemberVersion(5, 0, 0)
+_CLIENT_PUBLIC_ENDPOINT_QUALIFIER = EndpointQualifier(ProtocolType.CLIENT, "public")
+
+
+class ConnectionManagerTranslateTest(HazelcastTestCase):
+
+    rc = None
+    cluster = None
+    member = None
+
+    @classmethod
+    def setUpClass(cls):
+        cls.rc = cls.create_rc()
+        cls.cluster = cls.create_cluster(cls.rc, None)
+        cls.member = cls.cluster.start_member()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.rc.terminateCluster(cls.cluster.id)
+        cls.rc.exit()
+
+    def setUp(self):
+        self.client = None
+
+    def tearDown(self):
+        if self.client:
+            self.client.shutdown()
+
+    def test_translate_is_used(self):
+        # It shouldn't be able to connect to cluster using unreachable
+        # public address.
+        with self.assertRaises(IllegalStateError):
+            with patch.object(
+                HazelcastClient,
+                "_create_address_provider",
+                return_value=TestAddressProvider(True, self.member.address),
+            ):
+                self.client = HazelcastClient(
+                    cluster_name=self.cluster.id,
+                    cluster_connect_timeout=1.0,
+                    connection_timeout=1.0,
+                )
+
+    def test_translate_is_not_used_when_getting_existing_connection(self):
+        provider = TestAddressProvider(False, self.member.address)
+        with patch.object(
+            HazelcastClient,
+            "_create_address_provider",
+            return_value=provider,
+        ):
+            self.client = HazelcastClient(
+                cluster_name=self.cluster.id,
+                cluster_connect_timeout=1.0,
```

Comment:
Not really important, as it does not affect the behavior we test. So, I will leave it like this

Reply:
Actually I dont get it why it is used at all. Only testTranslateIsUsed uses a timeout. It can lead to test failures if connecting to the cluster takes more than 1 second. Can't it?

Reply:
This timeout is only considered when the first connection to the cluster fails. So, it can take longer than that.

Not really important, but let me remove them